components/dashboard/src/provider-utils.tsx

```diff
@@ -39,21 +39,24 @@ interface OpenAuthorizeWindowParams {
     login?: boolean;
     host: string;
     scopes?: string[];
+    additionalScopes?: string[];
     onSuccess?: (payload?: string) => void;
     onError?: (error?: string) => void;
 }
 
 async function openAuthorizeWindow(params: OpenAuthorizeWindowParams) {
-    const { login, host, scopes, onSuccess, onError } = params;
+    const { login, host, scopes, additionalScopes, onSuccess, onError } = params;
     const returnTo = gitpodHostUrl.with({ pathname: 'complete-auth', search: 'message=success' }).toString();
+    const requestedScopes = scopes || additionalScopes || [];
+    const overrideScopes = !!scopes;
```

Comment:
This API seems a bit weird: If you provide both `additionalScopes` and `scopes`, it will pick `additionalScopes` but override?

Why introduce a different `additionalScopes` list in the first place? Maybe it's simpler to just make `overrideScopes` configurable?

Reply:
Agree.